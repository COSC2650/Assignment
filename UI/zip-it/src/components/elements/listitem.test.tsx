import ListItem, { ListItemProp } from './listitem';
import { configure, render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { describe, expect } from '@jest/globals';

jest.useFakeTimers();

configure({ adapter: new Adapter() });

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useLayoutEffect: jest.requireActual('react').useEffect,
}));

describe('ListItem', () => {
  const props: ListItemProp = {
    listingTitle: 'Test Title',
    listingDescription: 'Test Description',
    listingPrice: 100.0,
    listingQuantity: 10,
    listingID: '',
    listingPostCode: 0,
    listingCategory: '',
    listingCondition: '',
    listingAvailibility: '',
    listingType: ''
  };

  it('Should render correctly', () => {
    const component = render(<ListItem {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('Title should be correct', () => {
    const wrapper = render(<ListItem {...props} />);
    expect(wrapper.find('h1#heading').text()).toEqual(props.listingTitle);
  });

  it('Description should be correct', () => {
    const wrapper = render(<ListItem {...props} />);
    expect(wrapper.find('#contents').text()).toEqual(props.listingDescription);
  });
});
