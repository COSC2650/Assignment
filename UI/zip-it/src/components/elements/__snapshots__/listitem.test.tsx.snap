// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Should render correctly 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "chakra-stack css-fmbk1s",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "chakra-image css-gz3640",
          "src": "https://picsum.photos/100?random=1",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "chakra-stack css-a03k66",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "chakra-heading css-0",
                "id": "heading",
              },
              "children": Array [
                Node {
                  "data": "Test Title",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h1",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {
                  "class": "chakra-text css-0",
                  "id": "contents",
                },
                "children": Array [
                  Node {
                    "data": "Test Description",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "class": "chakra-text css-0",
                    "id": "postcode",
                  },
                  "children": Array [
                    Node {
                      "data": "Postcode - 0",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "class": "chakra-text css-0",
                      "id": "price",
                    },
                    "children": Array [
                      Node {
                        "data": "Price - 100",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "p",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "id": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "id": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            Node {
              "attribs": Object {
                "class": "chakra-text css-0",
                "id": "contents",
              },
              "children": Array [
                Node {
                  "data": "Test Description",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {
                  "class": "chakra-text css-0",
                  "id": "postcode",
                },
                "children": Array [
                  Node {
                    "data": "Postcode - 0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "class": "chakra-text css-0",
                    "id": "price",
                  },
                  "children": Array [
                    Node {
                      "data": "Price - 100",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "class": "chakra-heading css-0",
                  "id": "heading",
                },
                "children": Array [
                  Node {
                    "data": "Test Title",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h1",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            Node {
              "attribs": Object {
                "class": "chakra-text css-0",
                "id": "postcode",
              },
              "children": Array [
                Node {
                  "data": "Postcode - 0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {
                  "class": "chakra-text css-0",
                  "id": "price",
                },
                "children": Array [
                  Node {
                    "data": "Price - 100",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "class": "chakra-text css-0",
                  "id": "contents",
                },
                "children": Array [
                  Node {
                    "data": "Test Description",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "chakra-heading css-0",
                    "id": "heading",
                  },
                  "children": Array [
                    Node {
                      "data": "Test Title",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h1",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            Node {
              "attribs": Object {
                "class": "chakra-text css-0",
                "id": "price",
              },
              "children": Array [
                Node {
                  "data": "Price - 100",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "class": "chakra-text css-0",
                  "id": "postcode",
                },
                "children": Array [
                  Node {
                    "data": "Postcode - 0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "chakra-text css-0",
                    "id": "contents",
                  },
                  "children": Array [
                    Node {
                      "data": "Test Description",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "chakra-heading css-0",
                      "id": "heading",
                    },
                    "children": Array [
                      Node {
                        "data": "Test Title",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "h1",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "id": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "id": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "src": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "src": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "chakra-stack css-a03k66",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "chakra-heading css-0",
              "id": "heading",
            },
            "children": Array [
              Node {
                "data": "Test Title",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h1",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "chakra-text css-0",
                "id": "contents",
              },
              "children": Array [
                Node {
                  "data": "Test Description",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {
                  "class": "chakra-text css-0",
                  "id": "postcode",
                },
                "children": Array [
                  Node {
                    "data": "Postcode - 0",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "class": "chakra-text css-0",
                    "id": "price",
                  },
                  "children": Array [
                    Node {
                      "data": "Price - 100",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "chakra-text css-0",
              "id": "contents",
            },
            "children": Array [
              Node {
                "data": "Test Description",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "chakra-text css-0",
                "id": "postcode",
              },
              "children": Array [
                Node {
                  "data": "Postcode - 0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {
                  "class": "chakra-text css-0",
                  "id": "price",
                },
                "children": Array [
                  Node {
                    "data": "Price - 100",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "chakra-heading css-0",
                "id": "heading",
              },
              "children": Array [
                Node {
                  "data": "Test Title",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h1",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "chakra-text css-0",
              "id": "postcode",
            },
            "children": Array [
              Node {
                "data": "Postcode - 0",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "chakra-text css-0",
                "id": "price",
              },
              "children": Array [
                Node {
                  "data": "Price - 100",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "chakra-text css-0",
                "id": "contents",
              },
              "children": Array [
                Node {
                  "data": "Test Description",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "class": "chakra-heading css-0",
                  "id": "heading",
                },
                "children": Array [
                  Node {
                    "data": "Test Title",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h1",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "chakra-text css-0",
              "id": "price",
            },
            "children": Array [
              Node {
                "data": "Price - 100",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "chakra-text css-0",
                "id": "postcode",
              },
              "children": Array [
                Node {
                  "data": "Postcode - 0",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "class": "chakra-text css-0",
                  "id": "contents",
                },
                "children": Array [
                  Node {
                    "data": "Test Description",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "chakra-heading css-0",
                    "id": "heading",
                  },
                  "children": Array [
                    Node {
                      "data": "Test Title",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h1",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "chakra-image css-gz3640",
            "src": "https://picsum.photos/100?random=1",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "src": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
